scipy/special/tests/test_data.py

```diff
@@ -430,6 +430,8 @@ def test_boost():
         data(spherical_jn_, 'sph_bessel_data_ipp-sph_bessel_data', (0,1), 2, rtol=1e-13),
         data(spherical_yn_, 'sph_neumann_data_ipp-sph_neumann_data', (0,1), 2, rtol=8e-15),
 
+        data(owens_t, 'owenst_data_ipp-owenst', (0, 1), 2, rtol=5e-1),
```

Comment:
Ok, you've quite ably defended against my accusations of bugs :-), so let's see whether the error should be that large. For the point which has large relative error: take the sum of the absolute value of the terms of the series and divide it by the sum of the series to get the condition number. Multiply the condition number by machine epsilon; if it's about the same size as the relative error then it's a mathematical difficulty (cancellation in an alternating series), and it's time to implement the extra methods in Boost. If it's much smaller then there's a bug.